language/tools/move-mv-llvm-compiler/src/stackless/translate.rs

```diff
@@ -848,6 +802,77 @@ impl<'mm, 'up> FunctionContext<'mm, 'up> {
         }
     }
 
+    fn collect_local_names(
+        &self,
+        fn_data: &FunctionData,
+        named_locals: &mut BTreeMap<mast::TempIndex, String>,
+    ) {
+        // Most locals in stackless bytecode are anonymous. We attempt here to collect and
+        // assign meaningful local names when they can be easily ascertained from the bytecode.
+        //
+        // A common and easly analyzable case is a where a local is involved in a structure
+        // operation. Direct examination of those operations yield field names which can
+        // then be mapped to the corresponding local. Consider:
+        //
+        // 0: $t1 := move($t0)
+        // 1: ($t2, $t3, $t4) := unpack Country::Country($t1)
+        // ...
+        //
+        // Above, $tN are anonymous names representing each local. By examining the unpack, it
+        // is straightforward to assign names to each local in turn from the structure referenced
+        // in the unpack operation. Similarly for other structure operations:
+        // - Locals targeted by Operation::Unpack.
+        // - Locals consumed by Operation::Pack.
+        // - Local extracted by Operation::BorrowField.
+        //
+        for instr in &fn_data.code {
+            use sbc::Operation;
+            if let sbc::Bytecode::Call(_, dst, op, src, None) = instr {
+                match op {
+                    Operation::BorrowField(mod_id, struct_id, _types, offset) => {
+                        assert_eq!(src.len(), 1);
+                        assert_eq!(dst.len(), 1);
+                        let senv = self
+                            .get_global_env()
+                            .get_module(*mod_id)
+                            .into_struct(*struct_id);
+                        let tmp_idx = dst[0];
+                        let fenv = senv.get_field_by_offset(*offset);
+                        let name = fenv.get_name().display(senv.symbol_pool()).to_string();
+                        named_locals.insert(tmp_idx, name);
+                    }
+                    Operation::Pack(mod_id, struct_id, _types) => {
+                        let senv = self
+                            .get_global_env()
+                            .get_module(*mod_id)
+                            .into_struct(*struct_id);
+                        assert_eq!(dst.len(), 1);
+                        assert_eq!(src.len(), senv.get_field_count());
+                        for (offset, tmp_idx) in src.iter().enumerate() {
+                            let fenv = senv.get_field_by_offset(offset);
+                            let name = fenv.get_name().display(senv.symbol_pool()).to_string();
+                            named_locals.insert(*tmp_idx, name);
+                        }
```

Comment:
```suggestion
pack_unpack_name!(mod_id, struct_id, src, dst, named_locals);
```
where `pack_unpack_name` defined like this
```
macro_rules!  pack_unpack_name {
    (
          $mod_id:expr, $struct_id:expr, $src:expr, $dst:expr, $named_locals:expr
   ) => {
                        let senv = self
                            .get_global_env()
                            .get_module(*$mod_id)
                            .into_struct(*$struct_id);
                        assert_eq!($dst.len(), 1);
                        assert_eq!($src.len(), senv.get_field_count());
                        for (offset, tmp_idx) in $src.iter().enumerate() {
                            let fenv = senv.get_field_by_offset(offset);
                            let name = fenv.get_name().display(senv.symbol_pool()).to_string();
                            $named_locals.insert(*tmp_idx, name);
                        };
}
```
and the Unpack case macro invocation would just have `src` and `dst` arguments flipped.
Up to you if you consider it would be beneficial to defined this macro.

Reply:
also mod_id, struct_id, and named_locals, don't need to be parameters of the macro, only `src` and `dst`